Output builtin

toni-calvin opened the pull request:
This PR aims to implement the output builtin, closes #86

Commit:
First commit

Commit:
Merge branch 'main' of github.com:lambdaclass/cairo-vm.go into output-builtin

Commit:
Initial structure

Commit:
Add basic methods

Commit:
Add basic methods

Commit:
Merge branch 'main' of github.com:lambdaclass/cairo-vm.go into output-builtin

Commit:
Added Output builtin unit tests

Commit:
Added Output builtin unit tests

Commit:
Fmt: New line eof

fmoletta reviewed (approved):
Note that the output builtin's behaviour (to output values to stdout) is present in the `write_output` function. This should be implemented before documenting the builtin (as it would be quite an empty section), but can be done in a follow-up pr.

fmoletta commented on pkg/builtins/output.go:
```diff
@@ -0,0 +1,41 @@
+	return &OutputBuiltinRunner{
+		base:     memory.NewRelocatable(0, 0),
+		included: included}
+}
```
```suggestion
func NewOutputBuiltinRunner(included bool) *OutputBuiltinRunner {
	return &OutputBuiltinRunner{included: included}
}
```
zero-values can be omitted during initialization

Commit:
Merge branch 'main' of github.com:lambdaclass/cairo-vm.go into output-builtin

Commit:
Integration test:
New file added: simple_print.cairo: Is a program that calculates 100/2 and add the result to the output buffer
Base code here: https://github.com/lambdaclass/cairo-vm/blob/030b7ebd9b9f435ea583aa2f66a47d0416833eaa/vm/src/tests/cairo_run_test.rs#L119-L122

toni-calvin committed:
Update pkg/builtins/output.go

Co-authored-by: fmoletta <99273364+fmoletta@users.noreply.github.com>
